### ISSUE TITLE
Planning for post release bug fixing

### ISSUE THREAD

**Issue by sjsrey** [pysal/pysal#106]
_Original author: sjsrey (July 01, 2010 02:35:06)_

The following is a proposal for how we maintain releases.

When 1.0 is released we do the following:
     create a tag 1.0
     create a branch 1.0
     bump the trunk up to version 1.1

Development on 1.1 happens in trunk. Code in tag 1.0 is never changed.
Bugs in 1.0 that are reported by users or developers are fixed in branch 1.0 and in trunk, if relevant. It could be that trunk had moved on so that the bug is no longer in the trunk code (functionality may have been dropped or refactored for example).

In the event that bug fixes are substantial between 1.0 and the release of 1.1, we could do a release out of branch 1.0. This would be tagged as 1.0.1. Since it is a tag, code in there never gets changed, but bug fixes in the 1.0.x series continue to be made in branch 1.0.

Bug fixes for 1.0 are handled until 2.0 is released (one year). 
We may want to have one developer oversee the bug fix process for a particular release. 

So to summarize our major versions two-dot:  1.0, 1.1, 2.0, 2.1, etc. These are released every six months at the end of a release cycle. Bug fix releases for a major release are three-dot: 2.0.1, 2.0.2, etc,  and are released during a release cycle.

_Original issue: http://code.google.com/p/pysal/issues/detail?id=106_
